### DISCUSSION TITLE
Redirect

### DISCUSSION THREAD

**Discussion by pikachu937** [bluenviron/mediamtx#2345] [Questions]
Hello.
I'm redirecting to another server via a regular expression for several threads:
```yaml
  "~^([0-9]{20})$":
    source: redirect
    sourceRedirect: rtsp://192.168.2.2:8554/$RTSP_PATH

```

but I can't capture them:

>2023/09/14 21:07:50 INF [RTSP] [conn 192.168.2.231:60156] opened
2023/09/14 21:07:50 DEB [RTSP] [conn 192.168.2.231:60156] [c->s] OPTIONS rtsp://192.168.2.3:8554/11111111111111111111 RTSP/1.0
CSeq: 2
User-Agent: LibVLC/3.0.18 (LIVE555 Streaming Media v2016.11.28)

>2023/09/14 21:07:50 DEB [RTSP] [conn 192.168.2.231:60156] [s->c] RTSP/1.0 200 OK
CSeq: 2
Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
Server: gortsplib

>2023/09/14 21:07:50 DEB [RTSP] [conn 192.168.2.231:60156] [c->s] DESCRIBE rtsp://192.168.2.3:8554/11111111111111111111 RTSP/1.0
Accept: application/sdp
CSeq: 3
User-Agent: LibVLC/3.0.18 (LIVE555 Streaming Media v2016.11.28)

>2023/09/14 21:07:50 DEB [RTSP] [conn 192.168.2.231:60156] [s->c] RTSP/1.0 401 Unauthorized
CSeq: 3
Server: gortsplib
WWW-Authenticate: Basic realm="IPCAM"

>2023/09/14 21:07:50 DEB [RTSP] [conn 192.168.2.231:60156] [c->s] DESCRIBE rtsp://192.168.2.3:8554/11111111111111111111 RTSP/1.0
Accept: application/sdp
Authorization: Basic emdla3Y4VNCYWVMTjVkb1FwejpMWFN3ekpzb3BU5pTm05RHVoSFpza3VMcmJjS1plOENhQGdSjc2x4empWOXZXSkRhcGpjSb2w5ZFVKORXVyaEpyQjZVS1haamtTZXJCcTFt
CSeq: 4
User-Agent: LibVLC/3.0.18 (LIVE555 Streaming Media v2016.11.28)

>2023/09/14 21:07:50 DEB [RTSP] [conn 192.168.2.231:60156] [s->c] RTSP/1.0 301 Moved Permanently
CSeq: 4
Location: rtsp://192.168.2.2:8554/$RTSP_PATH
Server: gortsplib

>2023/09/14 21:07:51 INF [RTSP] [conn 192.168.2.231:60156] closed (EOF)
2023/09/14 21:07:56 INF [RTSP] [conn 192.168.2.231:60164] opened
2023/09/14 21:07:56 DEB [RTSP] [conn 192.168.2.231:60164] [c->s] SETUP rtsp://192.168.2.3:8554/11111111111111111111 RTSP/1.0
CSeq: 0
Transport: RTP/AVP;unicast;client_port=9406-9407

>2023/09/14 21:07:56 INF [RTSP] [session 8c2b3be9] created by 192.168.2.231:60164
2023/09/14 21:07:56 DEB [RTSP] [conn 192.168.2.231:60164] [s->c] RTSP/1.0 400 Bad Request
CSeq: 0
Server: gortsplib

>2023/09/14 21:07:56 INF [RTSP] [conn 192.168.2.231:60164] closed (path of a SETUP request must end with a slash. This typically happens when VLC fails a request, and then switches to an unsupported RTSP dialect)
2023/09/14 21:07:56 INF [RTSP] [session 8c2b3be9] destroyed (not in use)
2023/09/14 21:07:56 INF [RTSP] [conn 192.168.2.231:60165] opened
2023/09/14 21:07:56 DEB [RTSP] [conn 192.168.2.231:60165] [c->s] OPTIONS rtsp://192.168.2.3:8554 RTSP/1.0
CSeq: 1
Clientchallenge: 9e26d33f2984236010ef6253fb1887f7
Clientid: Linux_2.4_6.0.9.1235_play32_RN01_EN_586
Companyid: KnKV4M4I/B2FjJ1TToLycw==
Guid: 00000000-0000-0000-0000-000000000000
Playerstarttime: [28/03/2003:22:50:23 00:00]
Regiondata: 0
User-Agent: RealMedia Player Version 6.0.9.1235 (linux-2.0-libc6-i386-gcc2.95)

>2023/09/14 21:07:56 DEB [RTSP] [conn 192.168.2.231:60165] [s->c] RTSP/1.0 200 OK
CSeq: 1
Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
Server: gortsplib

>2023/09/14 21:07:56 INF [RTSP] [conn 192.168.2.231:60165] closed (EOF)

but when I specify the full path for the redirect everything works
```yaml
  "~^([0-9]{20})$":
    source: redirect
    sourceRedirect: rtsp://192.168.2.2:8554/11111111111111111111

```

how can I redirect multiple threads without specifying the end thread?
Thank you

**Comment by aler9** [comment]
using variables inside `sourceRedirect` is currently unsupported but it may be in the future. Moved here: #2608
